<a href="https://colab.research.google.com/github/virf96/Neural-Networks/blob/main/Neurona_Clasificaci%C3%B3n.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from sklearn.datasets import load_breast_cancer
import pandas as pd
import numpy as np 

In [ ]:
cancer = load_breast_cancer()

In [ ]:
cancer.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename'])

In [ ]:
X,Y = cancer["data"], cancer["target"]

In [ ]:
X.shape

(569, 30)

In [ ]:
Y.shape

(569,)

### Categorias de la target

In [ ]:
np.unique(Y)

array([0, 1])

In [ ]:
type(X)

numpy.ndarray

In [ ]:
cancer["target_names"]

array(['malignant', 'benign'], dtype='<U9')

### Como DataFrame

In [ ]:
X_df = pd.DataFrame(X,columns=cancer["feature_names"])

In [ ]:
X_df.head()

mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0        17.99         10.38          122.80     1001.0          0.11840   
1        20.57         17.77          132.90     1326.0          0.08474   
2        19.69         21.25          130.00     1203.0          0.10960   
3        11.42         20.38           77.58      386.1          0.14250   
4        20.29         14.34          135.10     1297.0          0.10030   

   mean compactness  mean concavity  mean concave points  mean symmetry  \
0           0.27760          0.3001              0.14710         0.2419   
1           0.07864          0.0869              0.07017         0.1812   
2           0.15990          0.1974              0.12790         0.2069   
3           0.28390          0.2414              0.10520         0.2597   
4           0.13280          0.1980              0.10430         0.1809   

   mean fractal dimension  ...  worst radius  worst texture  worst perimeter  \
0                 0.07871  ...         25.38          17.33           184.60   
1                 0.05667  ...         24.99          23.41           158.80   
2                 0.05999  ...         23.57          25.53           152.50   
3                 0.09744  ...         14.91          26.50            98.87   
4                 0.05883  ...         22.54          16.67           152.20   

   worst area  worst smoothness  worst compactness  worst concavity  \
0      2019.0            0.1622             0.6656           0.7119   
1      1956.0            0.1238             0.1866           0.2416   
2      1709.0            0.1444             0.4245           0.4504   
3       567.7            0.2098             0.8663           0.6869   
4      1575.0            0.1374             0.2050           0.4000   

   worst concave points  worst symmetry  worst fractal dimension  
0                0.2654          0.4601                  0.11890  
1                0.1860          0.2750                  0.08902  
2                0.2430          0.3613                  0.08758  
3                0.2575          0.6638                  0.17300  
4                0.1625          0.2364                  0.07678  

[5 rows x 30 columns]

Dimensiones del data frame 

In [ ]:
X = X_df.values

In [ ]:
X

array([[1.799e+01, 1.038e+01, 1.228e+02, ..., 2.654e-01, 4.601e-01,
        1.189e-01],
       [2.057e+01, 1.777e+01, 1.329e+02, ..., 1.860e-01, 2.750e-01,
        8.902e-02],
       [1.969e+01, 2.125e+01, 1.300e+02, ..., 2.430e-01, 3.613e-01,
        8.758e-02],
       ...,
       [1.660e+01, 2.808e+01, 1.083e+02, ..., 1.418e-01, 2.218e-01,
        7.820e-02],
       [2.060e+01, 2.933e+01, 1.401e+02, ..., 2.650e-01, 4.087e-01,
        1.240e-01],
       [7.760e+00, 2.454e+01, 4.792e+01, ..., 0.000e+00, 2.871e-01,
        7.039e-02]])

In [ ]:
X.shape

(569, 30)

In [ ]:
X = X.T 

In [ ]:
X.shape

(30, 569)

$\alpha = 0.0001$

In [ ]:
Y.shape

(569,)

In [ ]:
Y = Y.reshape(1,-1)

In [ ]:
Y.shape

(1, 569)

In [ ]:
from sklearn.metrics import accuracy_score

Recuerden que la salida del modelo es una probabilidad, para las predicciones habria que hace run pre-procesamiento:
- si p>= .5 entonces y =1
- si p< 0.5 entonces y = 0

Si su modelos es "bueno" su funcion de costo deberia ser cercana a cero

### Escalamiento de datos 

In [ ]:
from sklearn.preprocessing import MinMaxScaler

In [ ]:
X.shape

(30, 569)

In [ ]:
mm = MinMaxScaler()
X = mm.fit_transform(X.T).T

In [ ]:
X.shape

(30, 569)

In [ ]:
def sigmoid(z):
    return 1/(1+np.exp(-z))

 m = Numero observaciones en nuestro dataset, en este caso es 569 \
 nx = numero de variables de nuestro dataset

In [ ]:
def initialize_with_zeros(nx):
    W = np.zeros((1,nx))
    b = 0.0
    return W,b 

In [ ]:
def propagate(W, b, X, Y):
    z = np.matmul(W,X)+ b
    A = sigmoid(z)
    m = A.shape[1]
    cost = -np.mean((np.multiply(Y,np.log(A))) + np.multiply(1-Y, np.log(1-A)))
    cost= np.squeeze(cost)
    dJ_dW = (1/m) * np.matmul(X,(A-Y).T)
    dJ_db = (1/m) * np.sum(A-Y)
    
    
    return cost, dJ_dW, dJ_db

In [ ]:
X.shape

(30, 569)

In [ ]:
probas = np.array([.1,.2,.4,.5,.6,.1,.8])

In [ ]:
np.where(probas>.5)

(array([4, 6], dtype=int64),)

In [ ]:
def predict(W,b, X):
    z = np.matmul(W,X)+ b
    A = sigmoid(z)
    return np.where(A>=.5, 1, 0)

In [ ]:
from sklearn.metrics import accuracy_score

In [ ]:
real = np.array([1,1,0])
pred = np.array([1,0,1]).reshape(3,1)

In [ ]:
real.shape

(3,)

In [ ]:
pred.shape

(3, 1)

In [ ]:
accuracy_score([1,1,1],[1,1,0])

0.6666666666666666

In [ ]:
def model(X,Y, iterations, learning_rate):
    W,b = initialize_with_zeros(X.shape[0])
    
    costs = list()
    accuracies = [accuracy_score(Y,predict(W,b,X))]
    for i in range(iterations):
        cost, dJ_dW, dJ_db = propagate(W,b,X,Y)
        
        costs.append(cost)
        W = W - learning_rate * dJ_dW.T
        b = b - learning_rate * dJ_db
        ## Accuracy 
        predict_classes = predict(W,b,X)        
        accuracies.append(accuracy_score(Y.reshape(-1,1), predict_classes.reshape(-1,1)))
        print("Cost after iteration {}".format(i), costs[i], "Accuracy",accuracies[i] )
    
    return W,b, costs, accuracies

In [ ]:
W,b, costs, accuracies = model(X,Y, iterations= 10000, learning_rate=1)

Cost after iteration 0 0.6931471805599453 Accuracy 0.0
Cost after iteration 1 0.6638431254938184 Accuracy 0.6502636203866432
Cost after iteration 2 0.6385317622142676 Accuracy 0.7504393673110721
Cost after iteration 3 0.6153666183349246 Accuracy 0.804920913884007
Cost after iteration 4 0.5940843214512037 Accuracy 0.8558875219683656
Cost after iteration 5 0.5745011796680521 Accuracy 0.8699472759226714
Cost after iteration 6 0.5564503295773661 Accuracy 0.8787346221441125
Cost after iteration 7 0.539779843820727 Accuracy 0.8857644991212654
Cost after iteration 8 0.5243525854415546 Accuracy 0.8875219683655536
Cost after iteration 9 0.5100455776704921 Accuracy 0.8980667838312829
Cost after iteration 10 0.49674902939374543 Accuracy 0.9015817223198594
Cost after iteration 11 0.48436519058177624 Accuracy 0.9050966608084359
Cost after iteration 12 0.47280715836071885 Accuracy 0.9086115992970123
Cost after iteration 13 0.4619977097990491 Accuracy 0.9103690685413005
Cost after iteration 14 0.4518

Cost after iteration 240 0.15900855201918856 Accuracy 0.9630931458699473
Cost after iteration 241 0.1587753395542217 Accuracy 0.9630931458699473
Cost after iteration 242 0.15854349112079796 Accuracy 0.9630931458699473
Cost after iteration 243 0.15831299301078172 Accuracy 0.9630931458699473
Cost after iteration 244 0.15808383171275714 Accuracy 0.9630931458699473
Cost after iteration 245 0.15785599390838906 Accuracy 0.9630931458699473
Cost after iteration 246 0.15762946646886564 Accuracy 0.9630931458699473
Cost after iteration 247 0.15740423645142052 Accuracy 0.9630931458699473
Cost after iteration 248 0.15718029109593257 Accuracy 0.9630931458699473
Cost after iteration 249 0.1569576178216014 Accuracy 0.9630931458699473
Cost after iteration 250 0.1567362042236959 Accuracy 0.9630931458699473
Cost after iteration 251 0.15651603807037506 Accuracy 0.9630931458699473
Cost after iteration 252 0.15629710729957788 Accuracy 0.9630931458699473
Cost after iteration 253 0.15607940001598195 Accuracy 

Cost after iteration 482 0.12499150624600552 Accuracy 0.968365553602812
Cost after iteration 483 0.12490449271568624 Accuracy 0.968365553602812
Cost after iteration 484 0.12481773712389434 Accuracy 0.968365553602812
Cost after iteration 485 0.12473123820943027 Accuracy 0.968365553602812
Cost after iteration 486 0.12464499471983648 Accuracy 0.968365553602812
Cost after iteration 487 0.12455900541131817 Accuracy 0.968365553602812
Cost after iteration 488 0.12447326904866464 Accuracy 0.968365553602812
Cost after iteration 489 0.124387784405172 Accuracy 0.968365553602812
Cost after iteration 490 0.12430255026256622 Accuracy 0.968365553602812
Cost after iteration 491 0.12421756541092742 Accuracy 0.968365553602812
Cost after iteration 492 0.1241328286486148 Accuracy 0.968365553602812
Cost after iteration 493 0.12404833878219249 Accuracy 0.968365553602812
Cost after iteration 494 0.12396409462635628 Accuracy 0.968365553602812
Cost after iteration 495 0.12388009500386093 Accuracy 0.96836555360

Cost after iteration 727 0.10922289839472284 Accuracy 0.9736379613356766
Cost after iteration 728 0.10917502185151359 Accuracy 0.9736379613356766
Cost after iteration 729 0.10912724234343747 Accuracy 0.9736379613356766
Cost after iteration 730 0.10907955955050448 Accuracy 0.9736379613356766
Cost after iteration 731 0.10903197315418788 Accuracy 0.9736379613356766
Cost after iteration 732 0.10898448283741562 Accuracy 0.9736379613356766
Cost after iteration 733 0.10893708828456178 Accuracy 0.9736379613356766
Cost after iteration 734 0.1088897891814381 Accuracy 0.9736379613356766
Cost after iteration 735 0.10884258521528545 Accuracy 0.9736379613356766
Cost after iteration 736 0.10879547607476539 Accuracy 0.9736379613356766
Cost after iteration 737 0.10874846144995196 Accuracy 0.9736379613356766
Cost after iteration 738 0.1087015410323233 Accuracy 0.9736379613356766
Cost after iteration 739 0.1086547145147535 Accuracy 0.9736379613356766
Cost after iteration 740 0.1086079815915045 Accuracy 0

Cost after iteration 964 0.10003608542665034 Accuracy 0.9789103690685413
Cost after iteration 965 0.10000463699766474 Accuracy 0.9789103690685413
Cost after iteration 966 0.09997323748551863 Accuracy 0.9789103690685413
Cost after iteration 967 0.09994188676593808 Accuracy 0.9789103690685413
Cost after iteration 968 0.09991058471508327 Accuracy 0.9789103690685413
Cost after iteration 969 0.0998793312095468 Accuracy 0.9789103690685413
Cost after iteration 970 0.09984812612635147 Accuracy 0.9789103690685413
Cost after iteration 971 0.0998169693429486 Accuracy 0.9789103690685413
Cost after iteration 972 0.09978586073721601 Accuracy 0.9789103690685413
Cost after iteration 973 0.09975480018745608 Accuracy 0.9789103690685413
Cost after iteration 974 0.09972378757239408 Accuracy 0.9789103690685413
Cost after iteration 975 0.09969282277117604 Accuracy 0.9789103690685413
Cost after iteration 976 0.09966190566336688 Accuracy 0.9789103690685413
Cost after iteration 977 0.09963103612894898 Accuracy

Cost after iteration 1203 0.09367953564012234 Accuracy 0.9806678383128296
Cost after iteration 1204 0.09365701521689268 Accuracy 0.9806678383128296
Cost after iteration 1205 0.09363452307775927 Accuracy 0.9806678383128296
Cost after iteration 1206 0.09361205916419997 Accuracy 0.9806678383128296
Cost after iteration 1207 0.09358962341785883 Accuracy 0.9806678383128296
Cost after iteration 1208 0.09356721578054572 Accuracy 0.9806678383128296
Cost after iteration 1209 0.09354483619423566 Accuracy 0.9806678383128296
Cost after iteration 1210 0.09352248460106827 Accuracy 0.9806678383128296
Cost after iteration 1211 0.09350016094334705 Accuracy 0.9806678383128296
Cost after iteration 1212 0.09347786516353891 Accuracy 0.9806678383128296
Cost after iteration 1213 0.09345559720427353 Accuracy 0.9806678383128296
Cost after iteration 1214 0.09343335700834285 Accuracy 0.9806678383128296
Cost after iteration 1215 0.09341114451870039 Accuracy 0.9806678383128296
Cost after iteration 1216 0.0933889596

Cost after iteration 1437 0.08907480399593827 Accuracy 0.9806678383128296
Cost after iteration 1438 0.08905759655040685 Accuracy 0.9806678383128296
Cost after iteration 1439 0.08904040722183901 Accuracy 0.9806678383128296
Cost after iteration 1440 0.08902323597851847 Accuracy 0.9806678383128296
Cost after iteration 1441 0.0890060827888053 Accuracy 0.9806678383128296
Cost after iteration 1442 0.0889889476211358 Accuracy 0.9806678383128296
Cost after iteration 1443 0.08897183044402215 Accuracy 0.9806678383128296
Cost after iteration 1444 0.08895473122605234 Accuracy 0.9806678383128296
Cost after iteration 1445 0.08893764993588964 Accuracy 0.9806678383128296
Cost after iteration 1446 0.08892058654227272 Accuracy 0.9806678383128296
Cost after iteration 1447 0.08890354101401517 Accuracy 0.9806678383128296
Cost after iteration 1448 0.08888651332000545 Accuracy 0.9806678383128296
Cost after iteration 1449 0.08886950342920641 Accuracy 0.9806678383128296
Cost after iteration 1450 0.088852511310

Cost after iteration 1672 0.0854697436845152 Accuracy 0.9824253075571178
Cost after iteration 1673 0.08545606014074063 Accuracy 0.9824253075571178
Cost after iteration 1674 0.08544238895147459 Accuracy 0.9824253075571178
Cost after iteration 1675 0.0854287300980019 Accuracy 0.9824253075571178
Cost after iteration 1676 0.08541508356164661 Accuracy 0.9824253075571178
Cost after iteration 1677 0.0854014493237717 Accuracy 0.9824253075571178
Cost after iteration 1678 0.08538782736577895 Accuracy 0.9824253075571178
Cost after iteration 1679 0.08537421766910912 Accuracy 0.9824253075571178
Cost after iteration 1680 0.08536062021524146 Accuracy 0.9824253075571178
Cost after iteration 1681 0.08534703498569393 Accuracy 0.9824253075571178
Cost after iteration 1682 0.08533346196202299 Accuracy 0.9824253075571178
Cost after iteration 1683 0.08531990112582336 Accuracy 0.9824253075571178
Cost after iteration 1684 0.08530635245872818 Accuracy 0.9824253075571178
Cost after iteration 1685 0.0852928159424

Cost after iteration 1909 0.08253571136284935 Accuracy 0.9824253075571178
Cost after iteration 1910 0.08252450655296824 Accuracy 0.9824253075571178
Cost after iteration 1911 0.08251331056469226 Accuracy 0.9824253075571178
Cost after iteration 1912 0.08250212338626901 Accuracy 0.9824253075571178
Cost after iteration 1913 0.0824909450059677 Accuracy 0.9824253075571178
Cost after iteration 1914 0.08247977541207932 Accuracy 0.9824253075571178
Cost after iteration 1915 0.08246861459291623 Accuracy 0.9824253075571178
Cost after iteration 1916 0.08245746253681235 Accuracy 0.9824253075571178
Cost after iteration 1917 0.08244631923212305 Accuracy 0.9824253075571178
Cost after iteration 1918 0.08243518466722516 Accuracy 0.9824253075571178
Cost after iteration 1919 0.08242405883051679 Accuracy 0.9824253075571178
Cost after iteration 1920 0.08241294171041734 Accuracy 0.9824253075571178
Cost after iteration 1921 0.08240183329536756 Accuracy 0.9824253075571178
Cost after iteration 1922 0.08239073357

Cost after iteration 2137 0.08018853844299638 Accuracy 0.9824253075571178
Cost after iteration 2138 0.0801790781143346 Accuracy 0.9824253075571178
Cost after iteration 2139 0.08016962440298554 Accuracy 0.9824253075571178
Cost after iteration 2140 0.0801601773010579 Accuracy 0.9824253075571178
Cost after iteration 2141 0.08015073680067315 Accuracy 0.9824253075571178
Cost after iteration 2142 0.08014130289396601 Accuracy 0.9824253075571178
Cost after iteration 2143 0.08013187557308411 Accuracy 0.9824253075571178
Cost after iteration 2144 0.08012245483018805 Accuracy 0.9824253075571178
Cost after iteration 2145 0.08011304065745145 Accuracy 0.9824253075571178
Cost after iteration 2146 0.08010363304706074 Accuracy 0.9824253075571178
Cost after iteration 2147 0.08009423199121525 Accuracy 0.9824253075571178
Cost after iteration 2148 0.08008483748212723 Accuracy 0.9824253075571178
Cost after iteration 2149 0.08007544951202175 Accuracy 0.9824253075571178
Cost after iteration 2150 0.080066068073

Cost after iteration 2366 0.0781806173323605 Accuracy 0.9824253075571178
Cost after iteration 2367 0.07817248939204724 Accuracy 0.9824253075571178
Cost after iteration 2368 0.07816436655533855 Accuracy 0.9824253075571178
Cost after iteration 2369 0.07815624881673375 Accuracy 0.9824253075571178
Cost after iteration 2370 0.07814813617074033 Accuracy 0.9824253075571178
Cost after iteration 2371 0.0781400286118741 Accuracy 0.9824253075571178
Cost after iteration 2372 0.07813192613465898 Accuracy 0.9824253075571178
Cost after iteration 2373 0.07812382873362719 Accuracy 0.9824253075571178
Cost after iteration 2374 0.07811573640331909 Accuracy 0.9824253075571178
Cost after iteration 2375 0.07810764913828327 Accuracy 0.9824253075571178
Cost after iteration 2376 0.0780995669330764 Accuracy 0.9824253075571178
Cost after iteration 2377 0.07809148978226332 Accuracy 0.9824253075571178
Cost after iteration 2378 0.07808341768041699 Accuracy 0.9824253075571178
Cost after iteration 2379 0.0780753506221

Cost after iteration 2612 0.07632258361286619 Accuracy 0.9824253075571178
Cost after iteration 2613 0.07631556350907705 Accuracy 0.9824253075571178
Cost after iteration 2614 0.07630854737054946 Accuracy 0.9824253075571178
Cost after iteration 2615 0.0763015351934144 Accuracy 0.9824253075571178
Cost after iteration 2616 0.07629452697380826 Accuracy 0.9824253075571178
Cost after iteration 2617 0.07628752270787262 Accuracy 0.9824253075571178
Cost after iteration 2618 0.07628052239175437 Accuracy 0.9824253075571178
Cost after iteration 2619 0.0762735260216055 Accuracy 0.9824253075571178
Cost after iteration 2620 0.07626653359358349 Accuracy 0.9824253075571178
Cost after iteration 2621 0.07625954510385079 Accuracy 0.9824253075571178
Cost after iteration 2622 0.07625256054857527 Accuracy 0.9824253075571178
Cost after iteration 2623 0.07624557992392983 Accuracy 0.9824253075571178
Cost after iteration 2624 0.07623860322609272 Accuracy 0.9824253075571178
Cost after iteration 2625 0.076231630451

Cost after iteration 2841 0.07481140039479009 Accuracy 0.9824253075571178
Cost after iteration 2842 0.07480519681789873 Accuracy 0.9824253075571178
Cost after iteration 2843 0.07479899644181592 Accuracy 0.9824253075571178
Cost after iteration 2844 0.07479279926367495 Accuracy 0.9824253075571178
Cost after iteration 2845 0.07478660528061272 Accuracy 0.9824253075571178
Cost after iteration 2846 0.07478041448976959 Accuracy 0.9824253075571178
Cost after iteration 2847 0.07477422688828962 Accuracy 0.9824253075571178
Cost after iteration 2848 0.07476804247332051 Accuracy 0.9824253075571178
Cost after iteration 2849 0.07476186124201334 Accuracy 0.9824253075571178
Cost after iteration 2850 0.07475568319152295 Accuracy 0.9824253075571178
Cost after iteration 2851 0.07474950831900763 Accuracy 0.9824253075571178
Cost after iteration 2852 0.07474333662162937 Accuracy 0.9824253075571178
Cost after iteration 2853 0.0747371680965535 Accuracy 0.9824253075571178
Cost after iteration 2854 0.07473100274

Cost after iteration 3074 0.07344691381493158 Accuracy 0.9824253075571178
Cost after iteration 3075 0.07344138540563372 Accuracy 0.9824253075571178
Cost after iteration 3076 0.07343585961574903 Accuracy 0.9824253075571178
Cost after iteration 3077 0.07343033644311475 Accuracy 0.9824253075571178
Cost after iteration 3078 0.07342481588557061 Accuracy 0.9824253075571178
Cost after iteration 3079 0.07341929794095887 Accuracy 0.9824253075571178
Cost after iteration 3080 0.0734137826071243 Accuracy 0.9824253075571178
Cost after iteration 3081 0.07340826988191416 Accuracy 0.9824253075571178
Cost after iteration 3082 0.0734027597631782 Accuracy 0.9824253075571178
Cost after iteration 3083 0.07339725224876864 Accuracy 0.9824253075571178
Cost after iteration 3084 0.07339174733654032 Accuracy 0.9824253075571178
Cost after iteration 3085 0.0733862450243504 Accuracy 0.9824253075571178
Cost after iteration 3086 0.0733807453100587 Accuracy 0.9824253075571178
Cost after iteration 3087 0.07337524819152

Cost after iteration 3315 0.07218565988964501 Accuracy 0.9824253075571178
Cost after iteration 3316 0.07218070553663555 Accuracy 0.9824253075571178
Cost after iteration 3317 0.07217575334681092 Accuracy 0.9824253075571178
Cost after iteration 3318 0.07217080331852047 Accuracy 0.9824253075571178
Cost after iteration 3319 0.07216585545011546 Accuracy 0.9824253075571178
Cost after iteration 3320 0.0721609097399489 Accuracy 0.9824253075571178
Cost after iteration 3321 0.07215596618637549 Accuracy 0.9824253075571178
Cost after iteration 3322 0.07215102478775173 Accuracy 0.9824253075571178
Cost after iteration 3323 0.07214608554243607 Accuracy 0.9824253075571178
Cost after iteration 3324 0.07214114844878847 Accuracy 0.9824253075571178
Cost after iteration 3325 0.07213621350517084 Accuracy 0.9824253075571178
Cost after iteration 3326 0.07213128070994677 Accuracy 0.9824253075571178
Cost after iteration 3327 0.07212635006148156 Accuracy 0.9824253075571178
Cost after iteration 3328 0.07212142155

Cost after iteration 3558 0.07104169457251058 Accuracy 0.9824253075571178
Cost after iteration 3559 0.07103722123053598 Accuracy 0.9824253075571178
Cost after iteration 3560 0.07103274969781048 Accuracy 0.9824253075571178
Cost after iteration 3561 0.07102827997305315 Accuracy 0.9824253075571178
Cost after iteration 3562 0.07102381205498444 Accuracy 0.9824253075571178
Cost after iteration 3563 0.07101934594232606 Accuracy 0.9824253075571178
Cost after iteration 3564 0.07101488163380111 Accuracy 0.9824253075571178
Cost after iteration 3565 0.07101041912813374 Accuracy 0.9824253075571178
Cost after iteration 3566 0.07100595842404964 Accuracy 0.9824253075571178
Cost after iteration 3567 0.07100149952027561 Accuracy 0.9824253075571178
Cost after iteration 3568 0.07099704241553982 Accuracy 0.9824253075571178
Cost after iteration 3569 0.07099258710857172 Accuracy 0.9824253075571178
Cost after iteration 3570 0.07098813359810191 Accuracy 0.9824253075571178
Cost after iteration 3571 0.0709836818

Cost after iteration 3801 0.07000501665362797 Accuracy 0.9824253075571178
Cost after iteration 3802 0.07000094812690347 Accuracy 0.9824253075571178
Cost after iteration 3803 0.06999688113253484 Accuracy 0.9824253075571178
Cost after iteration 3804 0.06999281566951127 Accuracy 0.9824253075571178
Cost after iteration 3805 0.06998875173682317 Accuracy 0.9824253075571178
Cost after iteration 3806 0.06998468933346196 Accuracy 0.9824253075571178
Cost after iteration 3807 0.06998062845841972 Accuracy 0.9824253075571178
Cost after iteration 3808 0.06997656911068979 Accuracy 0.9824253075571178
Cost after iteration 3809 0.06997251128926618 Accuracy 0.9824253075571178
Cost after iteration 3810 0.0699684549931441 Accuracy 0.9824253075571178
Cost after iteration 3811 0.06996440022131949 Accuracy 0.9824253075571178
Cost after iteration 3812 0.06996034697278943 Accuracy 0.9824253075571178
Cost after iteration 3813 0.06995629524655181 Accuracy 0.9824253075571178
Cost after iteration 3814 0.06995224504

Cost after iteration 4043 0.06906288624698598 Accuracy 0.984182776801406
Cost after iteration 4044 0.0690591611540277 Accuracy 0.984182776801406
Cost after iteration 4045 0.0690554373741903 Accuracy 0.984182776801406
Cost after iteration 4046 0.069051714906664 Accuracy 0.984182776801406
Cost after iteration 4047 0.06904799375063946 Accuracy 0.984182776801406
Cost after iteration 4048 0.06904427390530837 Accuracy 0.984182776801406
Cost after iteration 4049 0.06904055536986285 Accuracy 0.984182776801406
Cost after iteration 4050 0.06903683814349604 Accuracy 0.984182776801406
Cost after iteration 4051 0.06903312222540146 Accuracy 0.984182776801406
Cost after iteration 4052 0.06902940761477364 Accuracy 0.984182776801406
Cost after iteration 4053 0.06902569431080757 Accuracy 0.984182776801406
Cost after iteration 4054 0.06902198231269921 Accuracy 0.984182776801406
Cost after iteration 4055 0.06901827161964506 Accuracy 0.984182776801406
Cost after iteration 4056 0.06901456223084229 Accuracy 

Cost after iteration 4285 0.0681979112389878 Accuracy 0.984182776801406
Cost after iteration 4286 0.0681944818723683 Accuracy 0.984182776801406
Cost after iteration 4287 0.06819105364205616 Accuracy 0.984182776801406
Cost after iteration 4288 0.06818762654739402 Accuracy 0.984182776801406
Cost after iteration 4289 0.06818420058772479 Accuracy 0.984182776801406
Cost after iteration 4290 0.06818077576239213 Accuracy 0.984182776801406
Cost after iteration 4291 0.06817735207074008 Accuracy 0.984182776801406
Cost after iteration 4292 0.06817392951211332 Accuracy 0.984182776801406
Cost after iteration 4293 0.06817050808585702 Accuracy 0.984182776801406
Cost after iteration 4294 0.06816708779131701 Accuracy 0.984182776801406
Cost after iteration 4295 0.06816366862783962 Accuracy 0.984182776801406
Cost after iteration 4296 0.0681602505947716 Accuracy 0.984182776801406
Cost after iteration 4297 0.06815683369146033 Accuracy 0.984182776801406
Cost after iteration 4298 0.06815341791725385 Accuracy

Cost after iteration 4521 0.06741873427434968 Accuracy 0.984182776801406
Cost after iteration 4522 0.06741555596146752 Accuracy 0.984182776801406
Cost after iteration 4523 0.06741237864376824 Accuracy 0.984182776801406
Cost after iteration 4524 0.06740920232070907 Accuracy 0.984182776801406
Cost after iteration 4525 0.06740602699174786 Accuracy 0.984182776801406
Cost after iteration 4526 0.06740285265634267 Accuracy 0.984182776801406
Cost after iteration 4527 0.06739967931395223 Accuracy 0.984182776801406
Cost after iteration 4528 0.06739650696403547 Accuracy 0.984182776801406
Cost after iteration 4529 0.06739333560605199 Accuracy 0.984182776801406
Cost after iteration 4530 0.0673901652394615 Accuracy 0.984182776801406
Cost after iteration 4531 0.06738699586372442 Accuracy 0.984182776801406
Cost after iteration 4532 0.06738382747830153 Accuracy 0.984182776801406
Cost after iteration 4533 0.06738066008265389 Accuracy 0.984182776801406
Cost after iteration 4534 0.06737749367624311 Accura

Cost after iteration 4766 0.06666853968885487 Accuracy 0.984182776801406
Cost after iteration 4767 0.06666558995385125 Accuracy 0.984182776801406
Cost after iteration 4768 0.06666264109291888 Accuracy 0.984182776801406
Cost after iteration 4769 0.06665969310560843 Accuracy 0.984182776801406
Cost after iteration 4770 0.0666567459914708 Accuracy 0.984182776801406
Cost after iteration 4771 0.0666537997500573 Accuracy 0.984182776801406
Cost after iteration 4772 0.06665085438091949 Accuracy 0.984182776801406
Cost after iteration 4773 0.06664790988360939 Accuracy 0.984182776801406
Cost after iteration 4774 0.0666449662576792 Accuracy 0.984182776801406
Cost after iteration 4775 0.06664202350268168 Accuracy 0.984182776801406
Cost after iteration 4776 0.06663908161816967 Accuracy 0.984182776801406
Cost after iteration 4777 0.06663614060369655 Accuracy 0.984182776801406
Cost after iteration 4778 0.06663320045881588 Accuracy 0.984182776801406
Cost after iteration 4779 0.06663026118308173 Accuracy

Cost after iteration 5007 0.06598194041759595 Accuracy 0.984182776801406
Cost after iteration 5008 0.06597918907029122 Accuracy 0.984182776801406
Cost after iteration 5009 0.06597643849778376 Accuracy 0.984182776801406
Cost after iteration 5010 0.06597368869969657 Accuracy 0.984182776801406
Cost after iteration 5011 0.06597093967565301 Accuracy 0.984182776801406
Cost after iteration 5012 0.06596819142527675 Accuracy 0.984182776801406
Cost after iteration 5013 0.06596544394819165 Accuracy 0.984182776801406
Cost after iteration 5014 0.06596269724402196 Accuracy 0.984182776801406
Cost after iteration 5015 0.06595995131239188 Accuracy 0.984182776801406
Cost after iteration 5016 0.06595720615292633 Accuracy 0.984182776801406
Cost after iteration 5017 0.06595446176525023 Accuracy 0.984182776801406
Cost after iteration 5018 0.06595171814898869 Accuracy 0.984182776801406
Cost after iteration 5019 0.06594897530376728 Accuracy 0.984182776801406
Cost after iteration 5020 0.0659462332292118 Accura

Cost after iteration 5238 0.06536622709509014 Accuracy 0.984182776801406
Cost after iteration 5239 0.06536364522927822 Accuracy 0.984182776801406
Cost after iteration 5240 0.06536106405784187 Accuracy 0.984182776801406
Cost after iteration 5241 0.06535848358045986 Accuracy 0.984182776801406
Cost after iteration 5242 0.06535590379681151 Accuracy 0.984182776801406
Cost after iteration 5243 0.0653533247065761 Accuracy 0.984182776801406
Cost after iteration 5244 0.06535074630943319 Accuracy 0.984182776801406
Cost after iteration 5245 0.06534816860506258 Accuracy 0.984182776801406
Cost after iteration 5246 0.06534559159314432 Accuracy 0.984182776801406
Cost after iteration 5247 0.06534301527335856 Accuracy 0.984182776801406
Cost after iteration 5248 0.06534043964538583 Accuracy 0.984182776801406
Cost after iteration 5249 0.06533786470890664 Accuracy 0.984182776801406
Cost after iteration 5250 0.06533529046360201 Accuracy 0.984182776801406
Cost after iteration 5251 0.06533271690915295 Accura

Cost after iteration 5477 0.06476821616552547 Accuracy 0.984182776801406
Cost after iteration 5478 0.06476579158799978 Accuracy 0.984182776801406
Cost after iteration 5479 0.06476336763389044 Accuracy 0.984182776801406
Cost after iteration 5480 0.06476094430292341 Accuracy 0.984182776801406
Cost after iteration 5481 0.06475852159482468 Accuracy 0.984182776801406
Cost after iteration 5482 0.06475609950932043 Accuracy 0.984182776801406
Cost after iteration 5483 0.0647536780461371 Accuracy 0.984182776801406
Cost after iteration 5484 0.06475125720500118 Accuracy 0.984182776801406
Cost after iteration 5485 0.06474883698563946 Accuracy 0.984182776801406
Cost after iteration 5486 0.06474641738777889 Accuracy 0.984182776801406
Cost after iteration 5487 0.06474399841114653 Accuracy 0.984182776801406
Cost after iteration 5488 0.06474158005546969 Accuracy 0.984182776801406
Cost after iteration 5489 0.06473916232047573 Accuracy 0.984182776801406
Cost after iteration 5490 0.06473674520589237 Accura

Cost after iteration 5720 0.0641967507020991 Accuracy 0.984182776801406
Cost after iteration 5721 0.06419446995179894 Accuracy 0.984182776801406
Cost after iteration 5722 0.0641921897631673 Accuracy 0.984182776801406
Cost after iteration 5723 0.0641899101359688 Accuracy 0.984182776801406
Cost after iteration 5724 0.06418763106996847 Accuracy 0.984182776801406
Cost after iteration 5725 0.06418535256493146 Accuracy 0.984182776801406
Cost after iteration 5726 0.06418307462062281 Accuracy 0.984182776801406
Cost after iteration 5727 0.06418079723680792 Accuracy 0.984182776801406
Cost after iteration 5728 0.06417852041325241 Accuracy 0.984182776801406
Cost after iteration 5729 0.06417624414972183 Accuracy 0.984182776801406
Cost after iteration 5730 0.06417396844598207 Accuracy 0.984182776801406
Cost after iteration 5731 0.0641716933017989 Accuracy 0.984182776801406
Cost after iteration 5732 0.06416941871693857 Accuracy 0.984182776801406
Cost after iteration 5733 0.06416714469116719 Accuracy 

Cost after iteration 5959 0.06366711577575614 Accuracy 0.984182776801406
Cost after iteration 5960 0.06366496288466063 Accuracy 0.984182776801406
Cost after iteration 5961 0.06366281050288303 Accuracy 0.984182776801406
Cost after iteration 5962 0.06366065863021964 Accuracy 0.984182776801406
Cost after iteration 5963 0.06365850726646692 Accuracy 0.984182776801406
Cost after iteration 5964 0.06365635641142138 Accuracy 0.984182776801406
Cost after iteration 5965 0.06365420606487965 Accuracy 0.984182776801406
Cost after iteration 5966 0.06365205622663848 Accuracy 0.984182776801406
Cost after iteration 5967 0.06364990689649468 Accuracy 0.984182776801406
Cost after iteration 5968 0.06364775807424533 Accuracy 0.984182776801406
Cost after iteration 5969 0.06364560975968751 Accuracy 0.984182776801406
Cost after iteration 5970 0.0636434619526185 Accuracy 0.984182776801406
Cost after iteration 5971 0.0636413146528355 Accuracy 0.984182776801406
Cost after iteration 5972 0.0636391678601361 Accuracy

Cost after iteration 6207 0.06314830426343558 Accuracy 0.9859402460456942
Cost after iteration 6208 0.06314627173049629 Accuracy 0.9859402460456942
Cost after iteration 6209 0.0631442396597972 Accuracy 0.9859402460456942
Cost after iteration 6210 0.06314220805116152 Accuracy 0.9859402460456942
Cost after iteration 6211 0.06314017690441268 Accuracy 0.9859402460456942
Cost after iteration 6212 0.06313814621937434 Accuracy 0.9859402460456942
Cost after iteration 6213 0.0631361159958699 Accuracy 0.9859402460456942
Cost after iteration 6214 0.06313408623372328 Accuracy 0.9859402460456942
Cost after iteration 6215 0.0631320569327582 Accuracy 0.9859402460456942
Cost after iteration 6216 0.06313002809279854 Accuracy 0.9859402460456942
Cost after iteration 6217 0.06312799971366845 Accuracy 0.9859402460456942
Cost after iteration 6218 0.0631259717951919 Accuracy 0.9859402460456942
Cost after iteration 6219 0.06312394433719323 Accuracy 0.9859402460456942
Cost after iteration 6220 0.06312191733949

Cost after iteration 6451 0.06266566107946163 Accuracy 0.9859402460456942
Cost after iteration 6452 0.06266373632229337 Accuracy 0.9859402460456942
Cost after iteration 6453 0.06266181198702829 Accuracy 0.9859402460456942
Cost after iteration 6454 0.0626598880735119 Accuracy 0.9859402460456942
Cost after iteration 6455 0.06265796458158979 Accuracy 0.9859402460456942
Cost after iteration 6456 0.06265604151110746 Accuracy 0.9859402460456942
Cost after iteration 6457 0.06265411886191075 Accuracy 0.9859402460456942
Cost after iteration 6458 0.06265219663384547 Accuracy 0.9859402460456942
Cost after iteration 6459 0.0626502748267575 Accuracy 0.9859402460456942
Cost after iteration 6460 0.0626483534404928 Accuracy 0.9859402460456942
Cost after iteration 6461 0.06264643247489744 Accuracy 0.9859402460456942
Cost after iteration 6462 0.06264451192981753 Accuracy 0.9859402460456942
Cost after iteration 6463 0.06264259180509933 Accuracy 0.9859402460456942
Cost after iteration 6464 0.0626406721005

Cost after iteration 6697 0.06220451808172152 Accuracy 0.9859402460456942
Cost after iteration 6698 0.06220269265113515 Accuracy 0.9859402460456942
Cost after iteration 6699 0.0622008676068023 Accuracy 0.9859402460456942
Cost after iteration 6700 0.06219904294858704 Accuracy 0.9859402460456942
Cost after iteration 6701 0.06219721867635366 Accuracy 0.9859402460456942
Cost after iteration 6702 0.06219539478996659 Accuracy 0.9859402460456942
Cost after iteration 6703 0.06219357128929021 Accuracy 0.9859402460456942
Cost after iteration 6704 0.062191748174189015 Accuracy 0.9859402460456942
Cost after iteration 6705 0.06218992544452772 Accuracy 0.9859402460456942
Cost after iteration 6706 0.062188103100170734 Accuracy 0.9859402460456942
Cost after iteration 6707 0.06218628114098292 Accuracy 0.9859402460456942
Cost after iteration 6708 0.06218445956682906 Accuracy 0.9859402460456942
Cost after iteration 6709 0.06218263837757391 Accuracy 0.9859402460456942
Cost after iteration 6710 0.062180817

Cost after iteration 6939 0.0617737200904435 Accuracy 0.9859402460456942
Cost after iteration 6940 0.06177198433028002 Accuracy 0.9859402460456942
Cost after iteration 6941 0.06177024892544189 Accuracy 0.9859402460456942
Cost after iteration 6942 0.06176851387580886 Accuracy 0.9859402460456942
Cost after iteration 6943 0.06176677918126083 Accuracy 0.9859402460456942
Cost after iteration 6944 0.061765044841677816 Accuracy 0.9859402460456942
Cost after iteration 6945 0.06176331085693972 Accuracy 0.9859402460456942
Cost after iteration 6946 0.06176157722692662 Accuracy 0.9859402460456942
Cost after iteration 6947 0.061759843951518595 Accuracy 0.9859402460456942
Cost after iteration 6948 0.06175811103059588 Accuracy 0.9859402460456942
Cost after iteration 6949 0.06175637846403868 Accuracy 0.9859402460456942
Cost after iteration 6950 0.06175464625172725 Accuracy 0.9859402460456942
Cost after iteration 6951 0.06175291439354196 Accuracy 0.9859402460456942
Cost after iteration 6952 0.061751182

Cost after iteration 7190 0.061348888916479015 Accuracy 0.9859402460456942
Cost after iteration 7191 0.061347238720520746 Accuracy 0.9859402460456942
Cost after iteration 7192 0.06134558885148509 Accuracy 0.9859402460456942
Cost after iteration 7193 0.061343939309265655 Accuracy 0.9859402460456942
Cost after iteration 7194 0.061342290093755865 Accuracy 0.9859402460456942
Cost after iteration 7195 0.061340641204849486 Accuracy 0.9859402460456942
Cost after iteration 7196 0.06133899264244017 Accuracy 0.9859402460456942
Cost after iteration 7197 0.06133734440642176 Accuracy 0.9859402460456942
Cost after iteration 7198 0.06133569649668794 Accuracy 0.9859402460456942
Cost after iteration 7199 0.06133404891313264 Accuracy 0.9859402460456942
Cost after iteration 7200 0.06133240165564976 Accuracy 0.9859402460456942
Cost after iteration 7201 0.06133075472413329 Accuracy 0.9859402460456942
Cost after iteration 7202 0.061329108118477096 Accuracy 0.9859402460456942
Cost after iteration 7203 0.0613

Cost after iteration 7443 0.060941535211995454 Accuracy 0.9859402460456942
Cost after iteration 7444 0.06093996446340723 Accuracy 0.9859402460456942
Cost after iteration 7445 0.06093839401633576 Accuracy 0.9859402460456942
Cost after iteration 7446 0.060936823870686364 Accuracy 0.9859402460456942
Cost after iteration 7447 0.060935254026364384 Accuracy 0.9859402460456942
Cost after iteration 7448 0.060933684483275415 Accuracy 0.9859402460456942
Cost after iteration 7449 0.060932115241324816 Accuracy 0.9859402460456942
Cost after iteration 7450 0.060930546300418116 Accuracy 0.9859402460456942
Cost after iteration 7451 0.06092897766046084 Accuracy 0.9859402460456942
Cost after iteration 7452 0.06092740932135876 Accuracy 0.9859402460456942
Cost after iteration 7453 0.06092584128301735 Accuracy 0.9859402460456942
Cost after iteration 7454 0.06092427354534234 Accuracy 0.9859402460456942
Cost after iteration 7455 0.06092270610823954 Accuracy 0.9859402460456942
Cost after iteration 7456 0.0609

Cost after iteration 7692 0.06055949436221443 Accuracy 0.9859402460456942
Cost after iteration 7693 0.06055799587429133 Accuracy 0.9859402460456942
Cost after iteration 7694 0.060556497665581896 Accuracy 0.9859402460456942
Cost after iteration 7695 0.06055499973600151 Accuracy 0.9859402460456942
Cost after iteration 7696 0.06055350208546543 Accuracy 0.9859402460456942
Cost after iteration 7697 0.06055200471388915 Accuracy 0.9859402460456942
Cost after iteration 7698 0.06055050762118789 Accuracy 0.9859402460456942
Cost after iteration 7699 0.060549010807277204 Accuracy 0.9859402460456942
Cost after iteration 7700 0.0605475142720727 Accuracy 0.9859402460456942
Cost after iteration 7701 0.06054601801548967 Accuracy 0.9859402460456942
Cost after iteration 7702 0.06054452203744384 Accuracy 0.9859402460456942
Cost after iteration 7703 0.06054302633785082 Accuracy 0.9859402460456942
Cost after iteration 7704 0.06054153091662619 Accuracy 0.9859402460456942
Cost after iteration 7705 0.060540035

Cost after iteration 7939 0.060197645948284535 Accuracy 0.9859402460456942
Cost after iteration 7940 0.06019621394175347 Accuracy 0.9859402460456942
Cost after iteration 7941 0.060194782194590284 Accuracy 0.9859402460456942
Cost after iteration 7942 0.06019335070671893 Accuracy 0.9859402460456942
Cost after iteration 7943 0.06019191947806323 Accuracy 0.9859402460456942
Cost after iteration 7944 0.06019048850854709 Accuracy 0.9859402460456942
Cost after iteration 7945 0.06018905779809446 Accuracy 0.9859402460456942
Cost after iteration 7946 0.060187627346629295 Accuracy 0.9859402460456942
Cost after iteration 7947 0.06018619715407559 Accuracy 0.9859402460456942
Cost after iteration 7948 0.06018476722035731 Accuracy 0.9859402460456942
Cost after iteration 7949 0.06018333754539857 Accuracy 0.9859402460456942
Cost after iteration 7950 0.06018190812912352 Accuracy 0.9859402460456942
Cost after iteration 7951 0.06018047897145618 Accuracy 0.9859402460456942
Cost after iteration 7952 0.0601790

Cost after iteration 8178 0.0598626060880351 Accuracy 0.9859402460456942
Cost after iteration 8179 0.059861233974068444 Accuracy 0.9859402460456942
Cost after iteration 8180 0.05985986210214064 Accuracy 0.9859402460456942
Cost after iteration 8181 0.0598584904721827 Accuracy 0.9859402460456942
Cost after iteration 8182 0.05985711908412576 Accuracy 0.9859402460456942
Cost after iteration 8183 0.059855747937900744 Accuracy 0.9859402460456942
Cost after iteration 8184 0.05985437703343889 Accuracy 0.9859402460456942
Cost after iteration 8185 0.059853006370671195 Accuracy 0.9859402460456942
Cost after iteration 8186 0.059851635949528956 Accuracy 0.9859402460456942
Cost after iteration 8187 0.0598502657699433 Accuracy 0.9859402460456942
Cost after iteration 8188 0.05984889583184548 Accuracy 0.9859402460456942
Cost after iteration 8189 0.05984752613516667 Accuracy 0.9859402460456942
Cost after iteration 8190 0.059846156679838204 Accuracy 0.9859402460456942
Cost after iteration 8191 0.05984478

Cost after iteration 8419 0.05953877122091343 Accuracy 0.9859402460456942
Cost after iteration 8420 0.05953745550621129 Accuracy 0.9859402460456942
Cost after iteration 8421 0.0595361400177035 Accuracy 0.9859402460456942
Cost after iteration 8422 0.05953482475532724 Accuracy 0.9859402460456942
Cost after iteration 8423 0.05953350971902012 Accuracy 0.9859402460456942
Cost after iteration 8424 0.059532194908719395 Accuracy 0.9859402460456942
Cost after iteration 8425 0.059530880324362614 Accuracy 0.9859402460456942
Cost after iteration 8426 0.05952956596588728 Accuracy 0.9859402460456942
Cost after iteration 8427 0.05952825183323072 Accuracy 0.9859402460456942
Cost after iteration 8428 0.05952693792633063 Accuracy 0.9859402460456942
Cost after iteration 8429 0.05952562424512452 Accuracy 0.9859402460456942
Cost after iteration 8430 0.05952431078954986 Accuracy 0.9859402460456942
Cost after iteration 8431 0.05952299755954442 Accuracy 0.9859402460456942
Cost after iteration 8432 0.059521684

Cost after iteration 8666 0.059220509962032764 Accuracy 0.9859402460456942
Cost after iteration 8667 0.05921924827405898 Accuracy 0.9859402460456942
Cost after iteration 8668 0.05921798679753014 Accuracy 0.9859402460456942
Cost after iteration 8669 0.059216725532389276 Accuracy 0.9859402460456942
Cost after iteration 8670 0.059215464478579505 Accuracy 0.9859402460456942
Cost after iteration 8671 0.0592142036360439 Accuracy 0.9859402460456942
Cost after iteration 8672 0.05921294300472576 Accuracy 0.9859402460456942
Cost after iteration 8673 0.05921168258456824 Accuracy 0.9859402460456942
Cost after iteration 8674 0.05921042237551444 Accuracy 0.9859402460456942
Cost after iteration 8675 0.059209162377507625 Accuracy 0.9859402460456942
Cost after iteration 8676 0.059207902590491156 Accuracy 0.9859402460456942
Cost after iteration 8677 0.05920664301440816 Accuracy 0.9859402460456942
Cost after iteration 8678 0.05920538364920201 Accuracy 0.9859402460456942
Cost after iteration 8679 0.059204

Cost after iteration 8911 0.0589175818296026 Accuracy 0.9859402460456942
Cost after iteration 8912 0.05891637029573994 Accuracy 0.9859402460456942
Cost after iteration 8913 0.058915158960003516 Accuracy 0.9859402460456942
Cost after iteration 8914 0.058913947822341474 Accuracy 0.9859402460456942
Cost after iteration 8915 0.05891273688270185 Accuracy 0.9859402460456942
Cost after iteration 8916 0.05891152614103294 Accuracy 0.9859402460456942
Cost after iteration 8917 0.058910315597282784 Accuracy 0.9859402460456942
Cost after iteration 8918 0.058909105251399564 Accuracy 0.9859402460456942
Cost after iteration 8919 0.058907895103331526 Accuracy 0.9859402460456942
Cost after iteration 8920 0.0589066851530267 Accuracy 0.9859402460456942
Cost after iteration 8921 0.05890547540043366 Accuracy 0.9859402460456942
Cost after iteration 8922 0.05890426584550037 Accuracy 0.9859402460456942
Cost after iteration 8923 0.05890305648817511 Accuracy 0.9859402460456942
Cost after iteration 8924 0.0589018

Cost after iteration 9157 0.05862539061197166 Accuracy 0.9859402460456942
Cost after iteration 9158 0.05862422629873037 Accuracy 0.9859402460456942
Cost after iteration 9159 0.058623062171402965 Accuracy 0.9859402460456942
Cost after iteration 9160 0.05862189822994183 Accuracy 0.9859402460456942
Cost after iteration 9161 0.058620734474299704 Accuracy 0.9859402460456942
Cost after iteration 9162 0.05861957090442915 Accuracy 0.9859402460456942
Cost after iteration 9163 0.05861840752028262 Accuracy 0.9859402460456942
Cost after iteration 9164 0.058617244321812895 Accuracy 0.9859402460456942
Cost after iteration 9165 0.058616081308972376 Accuracy 0.9859402460456942
Cost after iteration 9166 0.058614918481713976 Accuracy 0.9859402460456942
Cost after iteration 9167 0.058613755839990234 Accuracy 0.9859402460456942
Cost after iteration 9168 0.05861259338375376 Accuracy 0.9859402460456942
Cost after iteration 9169 0.058611431112957336 Accuracy 0.9859402460456942
Cost after iteration 9170 0.058

Cost after iteration 9404 0.05834333826005804 Accuracy 0.9859402460456942
Cost after iteration 9405 0.058342218466757904 Accuracy 0.9859402460456942
Cost after iteration 9406 0.058341098848147054 Accuracy 0.9859402460456942
Cost after iteration 9407 0.05833997940418196 Accuracy 0.9859402460456942
Cost after iteration 9408 0.05833886013481923 Accuracy 0.9859402460456942
Cost after iteration 9409 0.05833774104001537 Accuracy 0.9859402460456942
Cost after iteration 9410 0.058336622119726934 Accuracy 0.9859402460456942
Cost after iteration 9411 0.05833550337391049 Accuracy 0.9859402460456942
Cost after iteration 9412 0.05833438480252267 Accuracy 0.9859402460456942
Cost after iteration 9413 0.058333266405520015 Accuracy 0.9859402460456942
Cost after iteration 9414 0.05833214818285913 Accuracy 0.9859402460456942
Cost after iteration 9415 0.05833103013449671 Accuracy 0.9859402460456942
Cost after iteration 9416 0.058329912260389256 Accuracy 0.9859402460456942
Cost after iteration 9417 0.05832

Cost after iteration 9647 0.058076264247825075 Accuracy 0.9859402460456942
Cost after iteration 9648 0.05807518566003113 Accuracy 0.9859402460456942
Cost after iteration 9649 0.05807410723678705 Accuracy 0.9859402460456942
Cost after iteration 9650 0.05807302897805285 Accuracy 0.9859402460456942
Cost after iteration 9651 0.058071950883788574 Accuracy 0.9859402460456942
Cost after iteration 9652 0.058070872953954175 Accuracy 0.9859402460456942
Cost after iteration 9653 0.058069795188509736 Accuracy 0.9859402460456942
Cost after iteration 9654 0.058068717587415275 Accuracy 0.9859402460456942
Cost after iteration 9655 0.0580676401506309 Accuracy 0.9859402460456942
Cost after iteration 9656 0.058066562878116575 Accuracy 0.9859402460456942
Cost after iteration 9657 0.0580654857698324 Accuracy 0.9859402460456942
Cost after iteration 9658 0.05806440882573866 Accuracy 0.9859402460456942
Cost after iteration 9659 0.05806333204579519 Accuracy 0.9859402460456942
Cost after iteration 9660 0.058062

Cost after iteration 9890 0.057818913046295724 Accuracy 0.9859402460456942
Cost after iteration 9891 0.057817873298835386 Accuracy 0.9859402460456942
Cost after iteration 9892 0.05781683370658656 Accuracy 0.9859402460456942
Cost after iteration 9893 0.05781579426951229 Accuracy 0.9859402460456942
Cost after iteration 9894 0.05781475498757588 Accuracy 0.9859402460456942
Cost after iteration 9895 0.05781371586074031 Accuracy 0.9859402460456942
Cost after iteration 9896 0.0578126768889688 Accuracy 0.9859402460456942
Cost after iteration 9897 0.05781163807222436 Accuracy 0.9859402460456942
Cost after iteration 9898 0.057810599410470336 Accuracy 0.9859402460456942
Cost after iteration 9899 0.05780956090366981 Accuracy 0.9859402460456942
Cost after iteration 9900 0.05780852255178607 Accuracy 0.9859402460456942
Cost after iteration 9901 0.05780748435478217 Accuracy 0.9859402460456942
Cost after iteration 9902 0.05780644631262148 Accuracy 0.9859402460456942
Cost after iteration 9903 0.05780540

### Función de perdida

In [ ]:
import plotly.graph_objects as go

In [ ]:
trace = go.Scatter(x = np.arange(len(costs)), y= costs, mode="lines")
fig = go.Figure()
fig.add_traces([trace])
fig.update_layout(dict(title= "Funcion de coste"))
fig.show()

In [ ]:
trace = go.Scatter(x = np.arange(len(accuracies)), y= accuracies, mode="lines")
fig = go.Figure()
fig.add_traces([trace])
fig.update_layout(dict(title= "Accuracy"))
fig.show()